### ISSUE TITLE
CA1024: False positive for special method GetPinnableReference

### ISSUE THREAD

**Issue by mihe** [dotnet/roslyn-analyzers#4634]
### Analyzer

**Diagnostic ID**: [CA1024](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1024): Use properties where appropriate

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.1 (Latest)

### Describe the bug

Rule CA1024 doesn't treat `GetPinnableReference` (used for allowing the type to be `fixed`) as a special method name, like it does with `GetHashCode` or `GetEnumerator` and instead triggers a warning on such methods.

### Steps To Reproduce

1. Add some code looking something like this:
```cs
public unsafe class Foo {
    private int _bar;
    public ref int GetPinnableReference() { return ref _bar; }
    public void Baz() { fixed (int* bar = this) { } }
}
```
2. Enable the CA1024 diagnostic
3. Analyze the code
3. Note the warning regarding `GetPinnableReference` being better off as a property.
